src/oop/special_class_method/hard/q9/MainTest.java

```diff
@@ -0,0 +1,289 @@
+                incorrectIntValueMessage);
+    }
+
+    private static Stream<Arguments> doubleInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, 432.2),
+                Arguments.of(234.4, 655.3),
+                Arguments.of(654.45, 453.3),
+                Arguments.of(43.3, 654.2)
+
+        );
+    }
+
+    @Test
+    public void setLandMassIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(setDoubleMethodName, new Class<?>[]{double.class}), String.format(incorrectMethodDefinition, setDoubleMethodName, className));
+        assertTrue(testClass.hasModifier(setDoubleMethodName, new Class<?>[]{double.class}, "public"), String.format(incorrectModifierMessage, setDoubleMethodName, className));
+        assertTrue(testClass.hasReturnType(setDoubleMethodName, new Class<?>[]{double.class}, Void.TYPE), String.format(incorrectReturnType, setDoubleMethodName, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("doubleInputProvider")
+    public void earthClassHasCorrectSetLandMassMethod(double initialValue, double setValue) throws Throwable {
+        Object[][] arguments = {
+                {initialValue, double.class},
+                {0, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {setValue, double.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setDoubleMethodName, "method does not correctly set the value of the", doubleFieldName, "attribute.");
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(setDoubleMethodName, new Class<?>[]{double.class}), String.format(incorrectMethodDefinition, setDoubleMethodName, className));
+        assertTrue(testClass.hasModifier(setDoubleMethodName, new Class<?>[]{double.class}, "public"), String.format(incorrectModifierMessage, setDoubleMethodName, className));
+        assertTrue(testClass.hasReturnType(setDoubleMethodName, new Class<?>[]{double.class}, Void.TYPE), String.format(incorrectReturnType, setDoubleMethodName, className));
```

Comment:
These should come first. Also, if this is a parsons question I don't think we should have two separate tests, it can just be one.

Reply:
You can apply this for the other methods below